Explorer/Assets/DCL/Multiplayer/SDK/Systems/WriteAvatarEmoteCommandSystem.cs

```diff
@@ -0,0 +1,91 @@
+    {
+        private readonly ISceneStateProvider sceneStateProvider;
+        private readonly IECSToCRDTWriter ecsToCRDTWriter;
+        private readonly IComponentPool<PBAvatarEmoteCommand> componentPool;
+
+        public WriteAvatarEmoteCommandSystem(World world, IECSToCRDTWriter ecsToCRDTWriter, IComponentPool<PBAvatarEmoteCommand> componentPool, ISceneStateProvider sceneStateProvider) : base(world)
+        {
+            this.ecsToCRDTWriter = ecsToCRDTWriter;
+            this.componentPool = componentPool;
+            this.sceneStateProvider = sceneStateProvider;
+        }
+
+        protected override void Update(float t)
+        {
+            HandleComponentRemovalQuery(World);
+            UpdateAvatarEmoteCommandQuery(World);
+            CreateAvatarEmoteCommandQuery(World);
+        }
+
+        [Query]
+        [None(typeof(PBAvatarEmoteCommand), typeof(DeleteEntityIntention))]
+        private void CreateAvatarEmoteCommand(in Entity entity, ref PlayerSDKDataComponent playerSDKDataComponent)
+        {
+            if (playerSDKDataComponent.PlayingEmote == null) return;
+
+            PBAvatarEmoteCommand pbComponent = componentPool.Get();
+            pbComponent.IsDirty = true;
+            pbComponent.EmoteUrn = playerSDKDataComponent.PlayingEmote;
+            pbComponent.Loop = playerSDKDataComponent.LoopingEmote;
+
+            ecsToCRDTWriter.AppendMessage<PBAvatarEmoteCommand, PBAvatarEmoteCommand>(static (dispatchedPBComponent, pbComponent) =>
+            {
+                dispatchedPBComponent.IsDirty = true;
+                dispatchedPBComponent.EmoteUrn = pbComponent.EmoteUrn;
+                dispatchedPBComponent.Loop = pbComponent.Loop;
+            }, playerSDKDataComponent.CRDTEntity, (int)sceneStateProvider.TickNumber, pbComponent);
+
+            World.Add(entity, pbComponent, playerSDKDataComponent.CRDTEntity);
+        }
+
+        [Query]
+        [None(typeof(DeleteEntityIntention))]
+        private void UpdateAvatarEmoteCommand(in Entity entity, ref PlayerSDKDataComponent playerSDKDataComponent, ref PBAvatarEmoteCommand pbComponent)
+        {
+            if (!playerSDKDataComponent.IsDirty || playerSDKDataComponent.PlayingEmote == null || playerSDKDataComponent.PlayingEmote.Equals(pbComponent.EmoteUrn))
+                return;
+
+            pbComponent.IsDirty = true;
+            pbComponent.EmoteUrn = playerSDKDataComponent.PlayingEmote;
+            pbComponent.Loop = playerSDKDataComponent.LoopingEmote;
+
+            ecsToCRDTWriter.AppendMessage<PBAvatarEmoteCommand, PBAvatarEmoteCommand>(static (dispatchedPBComponent, pbComponent) =>
+            {
+                dispatchedPBComponent.IsDirty = true;
+                dispatchedPBComponent.EmoteUrn = pbComponent.EmoteUrn;
+                dispatchedPBComponent.Loop = pbComponent.Loop;
+            }, playerSDKDataComponent.CRDTEntity, (int)sceneStateProvider.TickNumber, pbComponent);
+
+            World.Set(entity, pbComponent);
```

Comment:
you already modify the component by reference (moreover it's a ref type), This instruction is fully redundant.